Crash: bitwise XOR

```c
_Static_assert((1 ^ - 1) == -2, "");
```

```
panic: index out of bounds: index 2, len 1
/Users/ehaas/local/zig-macos-x86_64-0.12.0-dev.3508+a6ed3e6d2/lib/std/math/big/int.zig:1974:36: 0x10a34123b in normalize (arocc)
        r.len = llnormalize(r.limbs[0..length]);
                                   ^
/Users/ehaas/local/zig-macos-x86_64-0.12.0-dev.3508+a6ed3e6d2/lib/std/math/big/int.zig:1283:24: 0x10a41c35e in bitXor (arocc)
            r.normalize(b.limbs.len + @intFromBool(a.positive != b.positive));
                       ^
/Users/ehaas/source/arocc/src/aro/Value.zig:749:25: 0x10a41bf85 in bitXor (arocc)
    result_bigint.bitXor(lhs_bigint, rhs_bigint);
                        ^
```

--- a/src/aro/Value.zig
+++ b/src/aro/Value.zig
@@ -739,9 +739,10 @@ pub fn bitXor(lhs: Value, rhs: Value, comp: *Compilation) !Value {
     const lhs_bigint = lhs.toBigInt(&lhs_space, comp);
     const rhs_bigint = rhs.toBigInt(&rhs_space, comp);
 
+    const extra = @intFromBool(lhs_bigint.positive != rhs_bigint.positive);
     const limbs = try comp.gpa.alloc(
         std.math.big.Limb,
-        @max(lhs_bigint.limbs.len, rhs_bigint.limbs.len),
+        @max(lhs_bigint.limbs.len, rhs_bigint.limbs.len) + extra,
     );
     defer comp.gpa.free(limbs);
     var result_bigint = std.math.big.int.Mutable{ .limbs = limbs, .positive = undefined, .len = undefined };

--- a/test/cases/binary expressions.c
+++ b/test/cases/binary expressions.c
@@ -72,6 +72,13 @@ int invalid_ptr_arithmetic(struct Foo *num) {
     return num - num;
 }
 
+_Static_assert((1 ^ -1) == -2, "");
+_Static_assert((0 ^ 0) == 0, "");
+_Static_assert((-1 ^ 0) == -1, "");
+_Static_assert((-1 ^ -1) == 0, "");
+_Static_assert((-2 ^ 2) == -4, "");
+
+
 #define EXPECTED_ERRORS "binary expressions.c:3:7: error: invalid operands to binary expression ('long' and 'float')" \
     "binary expressions.c:6:13: error: invalid operands to binary expression ('char' and 'int *')" \
     "binary expressions.c:8:9: error: invalid operands to binary expression ('void (*)(void)' and 'void')" \

